index.js

```diff
@@ -1,206 +1,229 @@
+  }
 
-    var memopts = args.memops || {
-        maxAge: 10 * 60 * 1000,
-        /* 10 minutes */
-        preFetch: 0.5,
-        /* recache every 5 minutes */
-        primitive: true,
-        promise: true
-    }
+  _randomArray(a) {
+    return a[Math.floor(Math.random() * a.length)]
+  }
 
-    this.args = assign({}, this.args, processArgs(config, args))
-    this.filters = assign({}, Provider.DefaultFilters, config.filters)
+  resolveStream(src) {
+    this._warnDefault('resolveStream', 'multiple languages')
 
-    var cacher = _make_cached.bind(this)
-    this.fetch  = cacher(this.fetch,  memopts)
-    this.detail = cacher(this.detail, memopts)
-}
+    return src
+  }
 
-Provider.DefaultFilters = {
-    genres: {
-        all:         'All',
-        action:      'Action',
-        adventure:   'Adventure',
-        animation:   'Animation',
-        biography:   'Biography',
-        comedy:      'Comedy',
-        crime:       'Crime',
-        documentary: 'Documentary',
-        drama:       'Drama',
-        family:      'Family',
-        fantasy:     'Fantasy',
-        filmNoir:    'Film-Noir',
-        history:     'History',
-        horror:      'Horror',
-        music:       'Music',
-        musical:     'Musical',
-        mystery:     'Mystery',
-        romance:     'Romance',
-        sciFi:       'Sci-Fi',
-        short:       'Short',
-        sport:       'Sport',
-        thriller:    'Thriller',
-        war:         'War',
-        western:     'Western'
-    },
-    sorters: {
-        popularity:  'Popularity',
-        trending:    'Trending',
-        lastAdded:   'Last Added',
-        year:        'Year',
-        title:       'Title',
-        rating:      'Rating'
-    }
-}
+  random() {
+    this._warnDefault('random', 'faster random')
 
-Provider.ArgType = {
-    ARRAY:   'BUTTER_PROVIDER_ARG_TYPE_ARRAY',
-    OBJECT:  'BUTTER_PROVIDER_ARG_TYPE_OBJECT',
-    STRING:  'BUTTER_PROVIDER_ARG_TYPE_STRING',
-    BOOLEAN: 'BUTTER_PROVIDER_ARG_TYPE_BOOLEAN',
-    NUMBER:  'BUTTER_PROVIDER_ARG_TYPE_NUMBER'
-}
+    const uniqueId = this.config.uniqueId
+    return this.fetch()
+      .then(({ results }) => this._randomArray(results))
+      .then(data => this.detail(data[uniqueId], data))
+  }
 
-Provider.ItemType = {
-    MOVIE: 'movie',
-    TVSHOW: 'tvshow'
-}
+  extractIds(items) {
+    this._warnDefault('extractIds')
 
-Provider.OrderType = {
-    ASC: 'asc',
-    DESC: 'desc',
-    NULL: null
-}
+    return items.results.map(r => r[this.config.uniqueId])
+  }
 
-Provider.SorterType = {
-    NAME: 'name',
-    RATING: 'rating',
-    POPULARITY: 'popularity',
-    NULL: null
-}
+  detail(id, oldData) {
+    this._warnDefault(
+      `detail: ${id}`, 'better performing fetch and detail calls'
+    )
 
-Provider.QualityType = {
-    DEFAULT: '0',
-    LOW: '480p',
-    MEDIUM: '720p',
-    HIGH: '1080p'
-}
+    return Promise.resolve(oldData)
+  }
+
+  toString(arg) {
+    return JSON.stringify(this)
```

Comment:
I don't know you tell me: https://github.com/butterproviders/butter-provider/blob/master/index.js#L202